astropy/io/fits/diff.py

```diff
@@ -134,9 +135,9 @@ def identical(self):
         return not any(getattr(self, attr) for attr in self.__dict__
                        if attr.startswith('diff_'))
 
-    def report(self, fileobj=None, indent=0, clobber=False):
-        """
-        Generates a text report on the differences (if any) between two
+    @deprecated_renamed_argument('clobber', 'overwrite', '1.3')
+    def report(self, fileobj=None, indent=0, overwrite=False):
+        """Generates a text report on the differences (if any) between two
```

Comment:
It's not so much about conventions, it's more that everyone has a different preference and if everyone would change the style this could well lead to some unintended edit-wars with a lot of unnecessary commit-conflicts.